### PR TITLE
[docs] add folder architecture doc

### PR DESCRIPTION
**Pull request by Weiko**
## Context
We did most of the folders refactoring so we can now write a doc to help the community to contribute to the project.

### CODE REVIEW

**Comment by Weiko** [comment]
@charlesBochet probably a lot to arrange and rephrase there, I'll let you have a look or we can improve in a future PR!

File: docs/docs/developer/frontend/folder-architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+In this guide, you will explore the details of our project directory and how it contributes to the organization and maintainability of Twenty.
+
+By following this folder architecture convention, it is easier to find the files related to specific features and ensure that the application is scalable and maintainable.
+
+```
+front
+└───modules
+│   └───module1
+│   │   └───components
```
Comment (charlesBochet) [inline docs/docs/developer/frontend/folder-architecture.mdx:17]: + constants
+ types
+ utils

Comment (charlesBochet) [reply docs/docs/developer/frontend/folder-architecture.mdx:17]: + submodule

Comment (charlesBochet) [reply docs/docs/developer/frontend/folder-architecture.mdx:17]: a word about Hooks folder

Comment (charlesBochet) [reply docs/docs/developer/frontend/folder-architecture.mdx:17]: internal subfolder

**Review by charlesBochet — APPROVED** [review]
Thank you!

**Pull request merged**